### ISSUE TITLE
Adding a column with not null and default value

### ISSUE THREAD

**Issue by lookfirst** [mikro-orm/mikro-orm#534]
**Describe the bug**
`@Property({ default: '' })` 

produces:

`alter table "Table" add column "column" varchar(255) not null default ;`

**Expected behavior**

`alter table "Table" add column "column" varchar(255) not null default "";`

**Additional context**

I'm trying to add a column with `schema:update`... just to get the column into an existing database of rows, I need to default it to an empty string '' to start with and then I can remove that by hand.

Maybe I'm doing this wrong.

**Versions**

| Dependency | Version |
| - | - |
| mikro-orm | 3.6.12 |
| your-driver | pg 8.0.3 |

**Comment by B4nan** [comment]
Try to wrap it in quotes

**Comment by lookfirst** [comment]
That works!

But of course, the real solution is for mikro-orm to do it for me. ;-)

**Comment by B4nan** [comment]
Well the default is used as is to support things like “now()”, i am not planning to change this.

**Comment by lookfirst** [comment]
I don't get it... internally you see an empty string. Why can't mikro generate an empty string?

**Comment by B4nan** [comment]
Ok agreed, empty string is doable, my point is in generall, 1 is not “1” :)

**Comment by lookfirst** [comment]
Nor would I ask it to be. =)

**Comment by B4nan** [comment]
So today I faced this problem myself, but with enum value, which was even less obvious:

```typescript
@Enum({ default: Role.USER })
role = Role.USER;
```

And wrapping this in quotes looks quite ugly. Will change this in v4, so `default` will be strictly typed to `string/number/boolean/null` and strings will really mean strings there, plus there will be `defaultRaw: string` that will be used as is. Breaking, but I think its needed, recently I saw a project that was also misusing the `default` option (with `new Date()`), so having strictly defined type there should help.
